projects/gui/src/tournamentsettingswidget.cpp

```diff
@@ -32,6 +32,73 @@ TournamentSettingsWidget::TournamentSettingsWidget(QWidget *parent)
 		ui->m_seedsSpin->setEnabled(checked);
 	});
 
+	// number of rounds limits repeats
+	connect(ui->m_roundsSpin, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+		[=](int value)
+	{
+		if (!ui->m_gauntletRadio->isChecked())
+			value = 1;
+		int limit = value * ui->m_gamesPerEncounterSpin->value();
+		limit = (limit > 999999 ) ? 999999 : limit;
+		ui->m_repeatSpin->setMaximum(limit);
+	});
+
+	// number of games per encounter limits repeats
+	connect(ui->m_gamesPerEncounterSpin,
+		static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), [=](int value)
+	{	int rounds = ui->m_gauntletRadio->isChecked() ?  ui->m_roundsSpin->value() : 1;
+		int limit = value * rounds;
+		limit = (limit > 999999 ) ? 999999 : limit;
+		ui->m_repeatSpin->setMaximum(limit);
+	});
+
+	// update repeats after rounds
+	connect(ui->m_roundsSpin,
+		static_cast<void (QSpinBox::*)()>(&QAbstractSpinBox::editingFinished), [=]()
+	{
+		int value = ui->m_repeatSpin->value();
+		ui->m_repeatSpin->valueChanged(value);
+	});
+
+	// update repeats after games per encounter
+	connect(ui->m_gamesPerEncounterSpin,
+		static_cast<void (QSpinBox::*)()>(&QAbstractSpinBox::editingFinished),[=]()
+	{
+		int value = ui->m_repeatSpin->value();
+		ui->m_repeatSpin->valueChanged(value);
+	});
+
+	// update repeats via rounds spin after tournament type changed
+	connect(ui->m_gauntletRadio,
+		static_cast<void (QRadioButton::*)(bool)>(&QAbstractButton::toggled),[=]()
+	{
+		int value = ui->m_roundsSpin->value();
+		ui->m_roundsSpin->valueChanged(value);
+	});
+
+	// repeats dependent on games per encounter, rounds and tournament type
+	connect(ui->m_repeatSpin,static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+		[=](int valueIn)
+	{
+		static int store = 1;
+		int value = valueIn;
+		int step = (ui->m_repeatSpin->value() > store) ? 1 : -1;
+		int games = ui->m_gamesPerEncounterSpin->value();
+		int rounds = 1;
+		if (ui->m_gauntletRadio->isChecked())
+			rounds = ui->m_roundsSpin->value();
+
+		while (games % value
```

Comment:
This is an infinite loop when `games == 1` and `rounds == 1` (ie. the default state when someone opens the Settings dialog)

Reply:
When `games==1` and `rounds ==1` we have the condition

`(1 % value && ( value % 1 || ( 1 * 1 ) % value) )`

This is `( 1 % value && ( false || 1 % value) )` , or simply `(1 % value)`

Because `value` will always start between 1 and `rounds * games` (the maximum is controlled by the values the `m_gamesPerEncounterSpin` and `m_roundsSpin` SpinBoxes) and under the condition of the while loop it cannot leave this range, so no infinite loop occurs.

Reply:
An infinite loop may indeed occur for gauntlet tournaments, because `rounds * games` may overrun. Protection needed.

Reply:
I restructured the logic in this method and removed two other connect methods.